generators/server/templates/_Jenkinsfile

```diff
@@ -35,13 +41,25 @@ node {
     }
 
     stage('backend tests') {
-        sh "./gradlew test"
+        try {
+            sh "./gradlew test"
+        } catch(err) {
+            throw err
+        } finally {
+            step([$class: 'JUnitResultArchiver', testResults: '**/target/test-results/karma/TESTS-*.xml'])
+        }
     }
 
 <%_ } _%>
 <%_ if (applicationType !== 'microservice') { _%>
     stage('frontend tests') {
-        sh "gulp test"
+        try {
+            sh "gulp test"
+        } catch(err) {
+            throw err
+        } finally {
+            step([$class: 'JUnitResultArchiver', testResults: '**/target/test-results/karma/TESTS-*.xml'])
```

Comment:
I have tested it with Gulp, the tests are reported correctly:

<img width="1079" alt="screen shot 2016-11-17 at 9 17 35 am" src="https://cloud.githubusercontent.com/assets/579318/20392643/c35d6ec6-aca6-11e6-84d9-3dcbc45a9d12.png">